test/app.test.js

```diff
@@ -0,0 +1,70 @@
+const expect = require('chai').expect;
+const chai = require('chai');
+const chaiHttp = require('chai-http');
+chai.use(chaiHttp);
+const app = require('../app.js').app;
+const heroes = require('../app.js').heroes;
+
+describe('hero router', () => {
+    describe('GET /', () => {
+        it('Should return all heroes', (done) => {
+            chai.request(app).get('/heroes').end((err, res) => {
+                expect(res.body[0].name).to.eq(heroes[0].name);
+                expect(res.body).to.have.lengthOf(3);
+                done();
+            });
+        });
+
+        it('Should add a hero', (done) => {
+            chai.request(app).post('/heroes').send({ id: 4, name: 'Hero4' }).end((err, res) => {
+                expect(heroes[3].name).to.eq('Hero4');
+                expect(heroes).to.have.lengthOf(4);
+                done();
+            });
+        });
+
+        it('Should delete a hero', (done) => {
+            chai.request(app).delete('/heroes?name=Hero4').end((err, res) => {
+                expect(res.body).to.have.lengthOf(3);
+                done();
+            });
+        });
+
+        it('Should not find hero to delete', (done) => {
```

Comment:
not a good test over all, why do you think it's enough when it returns 3 heroes? should have at least checked how many heroes there are beforehand